runtime/src/main/cpp/Runtime.cpp

```diff
@@ -76,15 +76,30 @@ inline bool isValidRuntime() {
 
 volatile int aliveRuntimesCount = 0;
 
+enum GlobalRuntimeStatus {
+    kGlobalRuntimeUninitialized = 0,
+    kGlobalRuntimeRunning,
+    kGlobalRuntimeShutdown,
+};
+
+volatile GlobalRuntimeStatus globalRuntimeStatus = kGlobalRuntimeUninitialized;
+
 RuntimeState* initRuntime() {
-  SetKonanTerminateHandler();
   RuntimeState* result = konanConstructInstance<RuntimeState>();
   if (!result) return kInvalidRuntime;
   RuntimeCheck(!isValidRuntime(), "No active runtimes allowed");
   ::runtimeState = result;
+
+  bool firstRuntime = atomicAdd(&aliveRuntimesCount, 1) == 1;
+  auto lastStatus = compareAndSwap(&globalRuntimeStatus, kGlobalRuntimeUninitialized, kGlobalRuntimeRunning);
+  if (lastStatus == kGlobalRuntimeShutdown) {
+      konan::consoleErrorf("Kotlin runtime was shut down. Cannot create new runtimes\n");
+      konan::abort();
```

Comment:
I'm still not sure we should crash in this case.
Current implementation probably works.